.ci/Jenkinsfile

```diff
@@ -0,0 +1,106 @@
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+@Library('apm@current') _
+
+pipeline {
+  agent none
+  environment {
+    REPO = 'apm'
+    BASE_DIR = "src/github.com/elastic/${env.REPO}"
+    NOTIFY_TO = credentials('notify-to')
+    JOB_GCS_BUCKET = credentials('gcs-bucket')
+    PIPELINE_LOG_LEVEL='INFO'
+  }
+  options {
+    timeout(time: 3, unit: 'HOURS')
+    buildDiscarder(logRotator(numToKeepStr: '5', artifactNumToKeepStr: '5'))
+    timestamps()
+    ansiColor('xterm')
+    disableResume()
+    durabilityHint('PERFORMANCE_OPTIMIZED')
+  }
+  stages {
+    stage('Initializing'){
+      agent { label 'linux && immutable' }
+      options { skipDefaultCheckout() }
+      environment {
+        HOME = "${env.WORKSPACE}"
+        PATH = "${env.PATH}:${env.WORKSPACE}/bin:${env.WORKSPACE}/${env.BASE_DIR}/.ci/scripts"
+      }
+      stages {
+        stage('Checkout'){
+          agent { label 'immutable && docker' }
+          options { skipDefaultCheckout() }
+          steps {
+            deleteDir()
+            dir("${BASE_DIR}"){
+              checkout scm
+              stash allowEmpty: true, name: 'source', useDefaultExcludes: false
+            }
+          }
+        }
+        stage('Send Pull Request with APM specs to the APM Agents'){
+          agent { label 'immutable && docker' }
+          options {
+            skipDefaultCheckout()
+            warnError('Pull Requests to APM agents failed')
+          }
+          steps {
+            unstash 'source'
+            dir("${BASE_DIR}"){
+              script {
+                def agents = readYaml(file: '.ci/.jenkins-agents.yml')
+                def parallelTasks = [:]
+                agents['agents'].each { agent ->
+                  parallelTasks["apm-agent-${agent.NAME}"] = generateStepForAgent(agent: "${agent.NAME}", featuresPath: "${agent.FEATURES_PATH}")
+                }
+
+                parallel(parallelTasks)
+              }
+            }
+          }
+        }
+      }
+    }
+  }
+  post {
+    cleanup {
+      notifyBuildResult()
+    }
+  }
+}
+
+def generateStepForAgent(Map params = [:]){
+  def agent = params.get('agent')
+  def featuresPath = params.get('featuresPath')
+  log(level: 'INFO', text: "agent=${agent} featuresPath=${featuresPath}")
+  return {
+    node('linux && immutable') {
+      try {
+        deleteDir()
+        dir("${BASE_DIR}"){
+          unstash 'source'
```

Comment:
Not sure why the unstash has to be done here, inside the dir, but in https://github.com/elastic/apm/pull/231/files#diff-8db37e8fcea0f1a8f2f39667e94ebcc4R64 it's done outside the dir.

Reply:
See the line 93, it does provision a new worker to run in parallel

Reply:
Solved in new commits